src/git.rs

```diff
@@ -46,7 +46,7 @@ fn authentication_callback<'a>(
     };
 }
 
-fn checkout(repo: &Repository, commit_id: &str) -> Result<(), Error> {
+fn checkout<'a>(repo: &'a Repository, commit_id: &'a str) -> Result<Object<'a>, Error> {
```

Comment:
OPT: guts feeling but I guess we can avoid lifetime annotations here. Use String, use cloned Repo. I guess it doesn't add a lot of runtime perfs hit while simplifying the codebase avoiding future lifetime annotations leekage.

Reply:
compiler complains if you remove the lifetime here